File: web/eventhive/src/features/notifications/notificationsApi.ts
import { apiFetch } from '../../shared/api/client'
import type { NotificationItem } from '../../shared/types'

export const notificationsApi = {
  getAll: () =>
    apiFetch<NotificationItem[]>('/api/notifications'),

  getUnreadCount: () =>
    apiFetch<{ count: number }>('/api/notifications/unread-count'),

  markAsRead: (id: number) =>
    apiFetch<void>(`/api/notifications/${id}/read`, { method: 'PATCH' }),

  markAllAsRead: () =>
    apiFetch<void>('/api/notifications/read-all', { method: 'PATCH' }),
}

File: mobile/app/src/main/java/com/resuera/eventhive/features/notifications/NotificationsApi.kt
package com.resuera.eventhive.features.notifications

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.PATCH
import retrofit2.http.Path

interface NotificationsApi {

    @GET("/api/notifications")
    fun getNotifications(): Call<List<NotificationResponse>>

    @GET("/api/notifications/unread-count")
    fun getUnreadCount(): Call<Map<String, Int>>

    @PATCH("/api/notifications/{id}/read")
    fun markAsRead(@Path("id") id: Long): Call<Void>

    @PATCH("/api/notifications/read-all")
    fun markAllAsRead(): Call<Void>
}